services/device-registry-file/src/main/java/org/eclipse/hono/deviceregistry/file/FileBasedDeviceBackend.java

```diff
@@ -206,20 +241,21 @@ public FileBasedDeviceBackend(
     @Override
     public Future<CredentialsResult<JsonObject>> get(final String tenantId, final String type, final String authId,
             final JsonObject clientContext, final Span span) {
-        return credentialsService.get(tenantId, type, authId, clientContext, span)
-                .compose(credentialsResult -> {
-                    if (credentialsResult.isNotFound()) {
-                        return AutoProvisioning.provisionIfEnabled(
-                                tenantId,
-                                authId,
-                                clientContext,
-                                this,
-                                this,
-                                tenantInformationService,
-                                span);
-                    }
-                    return Future.succeededFuture(credentialsResult);
-                });
+
+        return this.tenantInformationService.getTenant(tenantId, span)
```

Comment:
wouldn't it make more sense to only retrieve the tenant if we couldn't find the credentials *and* auto-provisioning is enabled? It seems that otherwise we do not need the tenant, do we?

Reply:
In `AbstractCredentialsService` the tenant is being retrieved already to check if tenant exists or not but but not in the case of file/MongoDB registries. Based on #2110, the MongoDB device registry in the future will be using `AbstractCredentialsService`. So I made these changes accordingly.